cfy_manager/components/base_component/base_component.py

```diff
@@ -82,3 +88,90 @@ def validate_dependencies(self):
             self.logger.debug(
                 'All prerequisites for {class_name} are met'.format(
                     class_name=self.__class__.__name__))
+
+    def use_supplied_certificates(self, cert_destination, key_destination,
+                                  ca_destination,
+                                  owner=CLOUDIFY_USER, group=CLOUDIFY_GROUP):
+        """
+            Use user-supplied certificates, after checking they're not
+            broken.
+        """
+        cert_src, key_src, ca_src, key_pass = check_certificates(
+            self.component_name,
+        )
+
+        if not any([cert_src, key_src, ca_src, key_pass]):
+            # No certificates supplied, so not using them
+            self.logger.debug('No user-supplied certificates were present.')
+            return False
+
+        if key_pass:
+            # TODO: Strip password from key
```

Comment:
Can use this: https://github.com/cloudify-cosmo/cloudify-manager-install/blob/master/cfy_manager/utils/certificates.py#L376

Reply:
Should be added in latest update.